com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -824,5 +824,24 @@ protected NetworkObject GetNetworkObject(ulong networkId)
         {
             return NetworkManager.SpawnManager.SpawnedObjects.TryGetValue(networkId, out NetworkObject networkObject) ? networkObject : null;
         }
+
+        public void OnDestroy()
+        {
+            // this seems odd to do here, but in fact especially in tests we can find ourselves
+            //  here without having called InitializedVariables, which causes problems if any
+            //  of those variables use native containers (e.g. NetworkList) as they won't be
+            //  registered here and therefore won't be cleaned up.
+            //
+            // we should study to understand the initialization patterns
+            if (!m_VarInit)
```

Comment:
Again, there is something weird on why - in tests anyway - we instantiate a NB band then ditch it before we ever call InitializeVariables.  But regardless, we are now saying that Netvars need a chance to be disposed, and this ensures we don't skip any.  There are certainly better ways to register the netvars than the scan in InitializeVariables which we can look at later which might eliminate the need for this